builtins.go

```diff
@@ -233,7 +233,7 @@ func primitiveEquals(e *evaluator, xp potentialValue, yp potentialValue) (value,
 		if err != nil {
 			return nil, err
 		}
-		return makeValueBoolean(left.value == right.value), nil
+		return makeValueBoolean(string(left.value) == string(right.value)), nil
```

Comment:
I think this needs to compare the two arrays.  Firstly it avoids the intermediate representation.  Also can you confirm that string([]rune) is a loss-less conversion (e.g. no canonicalization).

Reply:
Yes, I believe it's a lossless conversion:

> In Go, a string is in effect a read-only slice of bytes. If you're at all uncertain about what a slice of bytes is or how it works, please read the previous blog post; we'll assume here that you have.
>
> It's important to state right up front that a string holds arbitrary bytes. It is not required to hold Unicode text, UTF-8 text, or any other predefined format. As far as the content of a string is concerned, it is exactly equivalent to a slice of bytes.

Source: https://blog.golang.org/strings

> Go's approach to normalization
> 
> As mentioned in the strings blog post, Go does not guarantee that characters in a string are
> normalized. However, the go.text packages can compensate. 

Source: https://blog.golang.org/normalization

Reply:
done